### PR TITLE
5 get-comments-by-article_id

### PR DESCRIPTION
**Pull request by jwiszowaty**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -81,6 +82,36 @@ describe('GET /api/articles/:article_id', () => {
...
+        })
+    })
+    it('return status 400 when passed article_id is not integer', () => {
+        return request(app)
+        .get('/api/articles/not-an-id/comments')
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe("Invalid input")
+        })
+    })
```
Comment (itskatherine) [inline __tests__/app.test.js:147]: Great tests so far Jakub 🙌 
Need to add a test to check that you get a 200 when you get comments from an article that doesn't have any comments (such as article 2) 🕵️

File: db/app.js [hunk 2]
```diff
@@ -2,7 +2,12 @@ const express = require("express");
 const app = express();
 const { getTopics } = require("../controllers/topics_controller");
 const { getEndpoints } = require("../controllers/endpoints_controller");
+<<<<<<< HEAD
```
Comment (itskatherine) [inline db/app.js]: These are left over from a merge conflict resolution, remember to get rid 🗑️

File: playground.txt [hunk 3]
```diff
@@ -16,6 +16,30 @@ You are now connected to database "nc_news_test" as user "jakubwiszowaty".
...
 (13 rows)
 
+<<<<<<< HEAD
```
Comment (itskatherine) [inline playground.txt]: This playground.txt and .sql file shouldn't be included in the PR

**Review by itskatherine — CHANGES_REQUESTED** [review]
Hi Jakub, nice work so far - another test is needed though as I've outlined in my comments
Could you please add that test and once you've got it passing please resubmit this PR 🙂

File: __tests__/app.test.js [hunk 4]
```diff
@@ -105,4 +106,42 @@ describe('GET /api/articles', () => {
...
+                expect(Object.keys(comment)).toEqual(expect.arrayContaining(expectedKeys))
+            })
+        })
+    })
+    it('return status 200 even when there are no comments for an existing article', () => {
+        return request(app)
+        .get('/api/articles/2/comments')
+        .expect(200)
+        .then(({body}) => {
+            expect(body.msg).toBe('No comments found for this article');
```
Comment (itskatherine) [inline __tests__/app.test.js]: If the article has no comments, I wouldn't expect an error message like this, rather an empty array

File: models/comments_models.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+exports.selectCommentsByArticleId = async (article_id) => {
+    const result = await db.query(
+        `SELECT comment_id, votes, created_at, author, body, article_id FROM comments
+        WHERE article_id = $1
+        ORDER BY created_at DESC`,
+        [article_id]
+    );
+    if (result.rows.length === 0 && article_id <= articles.length) {
+        return Promise.reject({ status: 200, msg: 'No comments found for this article' })
```
Comment (itskatherine) [inline models/comments_models.js]: If I wanted to get all the comments for an article with comments, I'd expect to get back an array filled with comment objects
If I wanted to get all the comments for an article with no comments, I'd still expect an array, but it would be empty
🔍

File: playground.txt [hunk 6]
```diff
@@ -16,6 +16,30 @@ You are now connected to database "nc_news_test" as user "jakubwiszowaty".
...
+ comment_id |                                                                                         body                                                                                         | article_id |    author     | votes |         created_at         
+------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+---------------+-------+----------------------------
```
Comment (itskatherine) [inline playground.txt]: This still needs removing from the PR please!

**Review by itskatherine — COMMENTED** [review]
Hi Jakub, please can you take another look at this, and resubmit once you've addressed my comments 🙂

File: playground.txt (hunk 3 above)
Comment (jwiszowaty) [reply playground.txt]: I hope I've managed to remove it, but I can't see it in my VS code :/
